alws/crud/errata.py

```diff
@@ -1715,3 +1715,66 @@ async def reset_matched_errata_packages(record_id: str, session: AsyncSession):
         items_to_insert.extend(matching_packages)
     session.add_all(items_to_insert)
     await session.flush()
+
+
+async def reset_matched_errata_packages_threshold(
+    issued_date_str: str, session: AsyncSession
+):
+    issued_date = datetime.datetime.strptime(
+        issued_date_str, '%Y-%m-%d %H:%M:%S'
+    )
+
+    stmt = (
+        select(models.NewErrataRecord)
+        .where(models.NewErrataRecord.issued_date >= issued_date)
+        .where(
+            models.NewErrataRecord.release_status
+            == ErrataReleaseStatus.NOT_RELEASED
+        )
+        .options(
+            selectinload(models.NewErrataRecord.platform).selectinload(
+                models.Platform.repos
+            ),
+            selectinload(models.NewErrataRecord.packages).selectinload(
+                models.NewErrataPackage.albs_packages
+            ),
+        )
+        .with_for_update()
+    )
+
+    result = await session.execute(stmt)
+    records = result.scalars().all()
+    print(records)
+    if not records:
+        return f"No unreleased records found after {issued_date_str} including"
+
+    for record in records:
+        items_to_insert = []
+        search_params = prepare_search_params(record)
+        prod_repos_cache = await load_platform_packages(
```

Comment:
This is very huge operation, not sure we should keep the whole work in endpoint. Maybe search for entries here and if count > 0 then fire a dramatiq task to actually do the modification while responding in endpoint that work was scheduled?